api/v1alpha1/perconaservermysql_types.go

```diff
@@ -745,8 +745,8 @@ func (cr *PerconaServerMySQL) PMMEnabled(secret *corev1.Secret) bool {
 
 func (pmm *PMMSpec) HasSecret(secret *corev1.Secret) bool {
 	if secret.Data != nil {
-		_, ok := secret.Data[string(UserPMMServerKey)]
-		return ok
+		v, ok := secret.Data[string(UserPMMServerKey)]
+		return ok && len(v) > 0
```

Comment:
Is it possible that we delete this properly instead of just checking is it empty or not?

Reply:
We delete it in internal secret, but we should not delete it in user secret

Reply:
This is approved, but just want to see why should not remove it in user secret?

Reply:
In general, if the user will use `replace` instead of `apply` he should not face this issue at all.